### DISCUSSION TITLE
Override a monitored scope alarm for single resource?

### DISCUSSION THREAD

**Discussion by lsmarsden** [cdklabs/cdk-monitoring-constructs#379] [Q&A]
Is it possible to override an alarm for a single resource in a monitored scope?

Suppose I have a `MonitoringFacade` which monitors a stack that uses `addFaultRateAlarm(...)` to apply to all lambda functions. This default fault rate alarm is good for 99% of my lambda functions in the stack, but say I have one particular lambda function in my stack that is extra important, and it should have a stricter value on its fault rate alarm. I can choose to add a fault rate alarm on this lambda function, but there are two problems that I see with this:
- You must manually override the alarm name, otherwise there is a conflict as two alarms with the same name are attempted to be created
- Even overriding the alarm name, you end up with two alarms on the lambda function. The default alarm, and the stricter alarm. So you will end up with two alarms, which doesn't work well (suppose the non-default alarm was a less-strict fault rate?)

Is there any way to achieve this currently? I imagine it's difficult, if not impossible, due to the way the MonitoringFacade uses Aspects to monitor resources. There doesn't seem to be any guarantee that you can 'remove' an aspect and replace it with a different aspect in a particular case.

I'm wondering what other solutions could work around this issue. Putting the lambda function in a stack outside the monitoring scope would work, but hinders things if it needs to be referenced throughout the stack it was removed from.

**Comment by bestickley** [comment]
@lsmarsden, I think this is a good feature request. Maybe a `filter` function could be added to the `aspectProps` (second argument of `monitorScope`) that would receive the construct as an argument and would expect a boolean to be returned so that you could filter which constructs the monitoring settings are applied to?

Like:
```ts
const monitoring = new MonitoringFacade(this, "Facade", { ... });
monitoring.monitorScope(stack1, {
  ..., // 99% fault rate settings
  filter: (construct) => {
    // return true for lower priority lambdas
  }
);
monitoring.monitorScope(stack1, {
  ..., // 99.99% fault rate settings
  filter: (construct) => {
    // return true for higher priority lambda
  }
);
```
